assets/js/components/ReportTable.js

```diff
@@ -24,8 +24,24 @@ import get from 'lodash/get';
 import invariant from 'invariant';
 import PropTypes from 'prop-types';
 
-export default function ReportTable( props ) {
-	const { rows, columns, className, limit, zeroState: ZeroState } = props;
+/**
+ * Internal dependencies.
+ */
+import { useFeature } from '../hooks/useFeature';
+import GatheringDataNotice from './GatheringDataNotice';
+
+export default function ReportTable( {
+	rows,
+	columns,
+	className,
+	limit,
+	zeroState: ZeroState,
+	isGatheringData = false,
+} ) {
+	const zeroDataStatesEnabled = useFeature( 'zeroDataStates' );
+	if ( ! zeroDataStatesEnabled ) {
+		isGatheringData = false;
```

Comment:
I put that in the IB thinking that we will be eventually removing the feature flag soon and `isGatheringData` will not be used in other cases according to the ACs. But yeah, just in case we add more logic around `isGatheringData` in the future, it might be better to inline them everywhere.